lib/slavery/connection_holder.rb

```diff
@@ -4,19 +4,24 @@ class ConnectionHolder < ActiveRecord::Base
 
     class << self
       # for delayed activation
-      def activate
-        spec = ActiveRecord::Base.configurations[Slavery.spec_key]
+      def activate(spec_key = nil)
+        spec = ActiveRecord::Base.configurations[spec_key || Slavery.spec_key]
         raise Error.new('Slavery.spec_key invalid!') if spec.nil?
         establish_connection spec
       end
     end
   end
 
   class << self
-    def connection_holder
-      @connection_holder ||= begin
-        ConnectionHolder.activate
-        ConnectionHolder
+    def connection_holder(target,spec_key)
```

Comment:
Why both `target` and `spec_key`? Seems unnecessary complication to me. Can't you do with just `spec_key`?